xio-core/src/main/java/zipkin2/reporter/okhttp3/OkHttpClientBuilderFactory.java

```diff
@@ -0,0 +1,45 @@
+package zipkin2.reporter.okhttp3;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.OkHttpClient;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+
+public class OkHttpClientBuilderFactory {
+  public static OkHttpClient.Builder createZipkinClientBuilder() {
+    TrustManager trustManager = new ZipkinTrustManager();
+    TrustManager[] trustManagers = {trustManager};
+    SSLContext sslCtx = null;
+    try {
+      sslCtx = SSLContext.getInstance("TLSv1.2");
+      sslCtx.init(null, trustManagers, null);
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      System.exit(-1);
+      throw new RuntimeException(e);
+    }
+
+    return new OkHttpClient.Builder()
```

Comment:
maybe later we can have this be a parameter from config file? to disable host checkin

Reply:
This sounds like a great idea. I will create a ticket.